src/components/Courses/Courses.jsx

```diff
@@ -1,45 +1,113 @@
+import CourseInfo from '../CourseInfo/CourseInfo';
+import { api_endpoints, storageKeys, paths } from '../../constants';
+import './Courses.css';
 
-const Courses = ({ courses, authors, onAddCourseClick }) => {
+const Courses = ({ onAddCourseClick }) => {
+  const [courses, setCourses] = useState([]);
+  const [authors, setAuthors] = useState([]);
   const [searchTerm, setSearchTerm] = useState('');
+  const [selectedCourse, setSelectedCourse] = useState(null);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const token = localStorage.getItem(storageKeys.userToken);
+    if (token) {
+      navigate(paths.courses);
+    }
+    fetchCourses();
+    fetchAuthors();
+  }, [navigate]);
 
-  const handleSearch = (term) => {
+  const fetchCourses = async (query = '', type = '') => {
+    try {
+      let response;
+      if (query!=='' && type!=='') {
+        response = await fetch(`${api_endpoints.filter}${type}=${query}`);
+      }
+      else{
+        response = await fetch(`${api_endpoints.courses}`);
+      }
+      
+      if (response.ok) {
+        const data = await response.json();
+        setCourses(data.result);
+      } else {
+        console.error('Failed to fetch courses:', response.status);
+      }
+    } catch (error) {
+      console.error('Error fetching courses:', error);
+    }
+  };
+
+  const fetchAuthors = async () => {
+    try {
+      const response = await fetch(api_endpoints.authors);
+      if (response.ok) {
+        const data = await response.json();
+        setAuthors(data.result);
+      } else {
+        console.error('Failed to fetch authors:', response.status);
+      }
+    } catch (error) {
+      console.error('Error fetching authors:', error);
+    }
+  };
+  const handleSearch = async (term) => {
     setSearchTerm(term);
+    await fetchCourses(term, 'title');
+    if (courses.length === 0) {
+      await fetchCourses(term, 'id');
+    }
   };
 
-  const filteredCourses = useMemo(() => {
-    return courses.filter(course =>
-      course.title.toLowerCase().includes(searchTerm.toLowerCase())
-    );
-  }, [courses, searchTerm]);
+  const handleAddCourse = () => {
+    navigate(paths.addCourse);
+    onAddCourseClick();
+  };
+
+  const handleShowCourseInfo = (course) => {
+    setSelectedCourse(course);
+  };
+
+  const handleBackToCourses = () => {
+    setSelectedCourse(null);
+  };
 
   return (
     <div className="courses">
       <div className="courses__search-bar">
-        <SearchBar courses={courses} onSearch={handleSearch} />
-        <Button onClick={onAddCourseClick}>Add New Course</Button>
+        <SearchBar onSearch={handleSearch} />
+        <Button onClick={handleAddCourse}>Add New Course</Button>
       </div>
       <div className="courses__list">
-        {filteredCourses.map(course => (
-          <CourseCard
-            key={course.id}
-            course={course}
-            authors={authors}
+        {selectedCourse ? (
+          <CourseInfo 
```

Comment:
With this approach you're effectively showing the CourseInfo whenever you click on a course card.
Judging from the [task](https://d17btkcdsmqrmh.cloudfront.net/react-fundamentals/docs/module-2/home-task/task#courseinfo-component), it seems that you need to navigate to the CourseInfo as a separate page with a different pathname. And on that page you make a separate API call for this exact Course.
![image](https://github.com/JoseCarlosBG/ReactTasks/assets/25910474/07603531-d886-4c67-8c93-a33b30d4e496)